### ISSUE TITLE
Remove `auto` as a type for `accessibility.voice.autoSynthesize`

### ISSUE THREAD

**Issue by meganrogge** [microsoft/vscode#229403]
Feedback from screen reader users like @jooyoungseo suggests that we should enable this by default just like we do for sighted users.

**Comment by meganrogge** [comment]
Marked feature so it gets verified early on.

**Comment by meganrogge** [comment]
Verify that your setting gets migrated from `"on" | "off" | "auto"` to the proper `boolean` value. 

By default, this is enabled in insider's and disabled for stable.

If you have it set to `on` or `auto`, it should be `true`. Otherwise, `false`.

**Comment by meganrogge** [comment]
The migration part is tricky because the registration happens later because the setting is registered after some time when the speech provider is recognized. The `registerConfiguration` call seems to actually be async, as when I insert the migration code after it, it says the setting is unrecognized....

**Comment by bpasero** [comment]
@meganrogge help me understand: `auto` is still useful to have a way to forcefully enable this for users with screen readers, or not? If we remove `auto`, we will forcefully disable this feature for all stable users (because it is not on by default).

I think we can only remove `auto` if we make the default of this setting `true` and not just for insiders 🤔

**Comment by meganrogge** [comment]
Screen reader users have said they want the same behavior as sighted users - so, on by default in insider's.

I think this means we can remove `auto`, as that just means on by default for non-screen reader users and off by default for screen reader users.

Re insider's vs stable, however that's currently handled would stay the same, I think?

**Comment by bpasero** [comment]
I see, yes we can do that. I think originally we added `auto` for screen reader users and I am fine removing it again.

**Comment by meganrogge** [comment]
I explored doing that and encountered some challenges. Ideas would be appreciated 

https://github.com/microsoft/vscode/issues/229403#issuecomment-2368808890

**Comment by bpasero** [comment]
I am not sure, but we could also just treat `auto` as either `on` or `off` internally and thus soft-deprecate this setting until we remove it.

**Comment by doggy8088** [comment]
The `accessibility.voice.autoSynthesize` seems removed. I can't find this setting in my VSCode insiders.

```
version: 1.97.0-insider (user setup)
commit: ce50bd4876af457f64d83cfd956bc916535285f4
date: 2024-12-16T05:04:30.005Z
Electron: 32.2.6
ElectronBuildId: 10629634
Chromium: 128.0.6613.186
Node.js: 20.18.1
V8: 12.8.374.38-electron.0
OS: Windows_NT x64 10.0.19045
```
